### PR TITLE
Copter: ModeGuided: add an option_is_enabled method, use it

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
Added a test for one of the options, others done my inspection only

File: ArduCopter/mode_guided.cpp [hunk 1]
```diff
@@ -1009,7 +1014,7 @@ void ModeGuided::set_yaw_state(bool use_yaw, float yaw_cd, bool use_yaw_rate, fl
 // returns true if pilot's yaw input should be used to adjust vehicle's heading
 bool ModeGuided::use_pilot_yaw(void) const
 {
-    return (copter.g2.guided_options.get() & uint32_t(Options::IgnorePilotYaw)) == 0;
+    return option_is_enabled(Option::IgnorePilotYaw);
```
Comment (IamPete1) [inline ArduCopter/mode_guided.cpp:1017]:
```suggestion
    return !option_is_enabled(Option::IgnorePilotYaw);
```

**Review by IamPete1 — CHANGES_REQUESTED** [review]

File: ArduCopter/mode.h [hunk 2]
```diff
@@ -1138,6 +1138,8 @@ class ModeGuided : public Mode {
         AllowWeatherVaning = (1U << 7)
     };
 
+    bool option_is_enabled(Option option) const;
```
Comment (rmackay9) [inline ArduCopter/mode.h:1141]: let's put a little comment above this one.  sorry, it's perhaps pretty obvious what it does but..

File: ArduCopter/mode_guided.cpp [hunk 3]
```diff
@@ -97,6 +97,11 @@ void ModeGuided::run()
     }
  }
 
+bool ModeGuided::option_is_enabled(Option option) const
```
Comment (rmackay9) [inline ArduCopter/mode_guided.cpp:100]: put a comment above here too please!

**Comment by rmackay9** [comment]
feel free to merge after those couple of small fixes.

File: ArduCopter/mode.h (hunk 2 above)
Comment (peterbarker) [reply ArduCopter/mode.h:1142]: Done

File: ArduCopter/mode_guided.cpp (hunk 3 above)
Comment (peterbarker) [reply ArduCopter/mode_guided.cpp:101]: Done

**Comment by peterbarker** [comment]
I've confirmed that using the `copter` singleton in the mode header does not work trivially, and my guess is that we'll end up with an include loop if we try to make it work.

**Pull request merged**